packages/fluentui/react-northstar/src/components/Carousel/Carousel.tsx

```diff
@@ -128,104 +129,112 @@ export const carouselSlotClassNames: CarouselSlotClassNames = {
+      ariaRoleDescription,
+      ariaLabel,
     }),
-    activeIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    ariaRoleDescription: PropTypes.string,
-    ariaLabel: PropTypes.string,
-    circular: PropTypes.bool,
-    defaultActiveIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    getItemPositionText: PropTypes.func,
-    items: customPropTypes.collectionShorthand,
-    navigation: PropTypes.oneOfType([customPropTypes.collectionShorthand, customPropTypes.itemShorthand]),
-    navigationPosition: PropTypes.string,
-    onActiveIndexChange: PropTypes.func,
-    paddleNext: customPropTypes.itemShorthand,
-    paddlesPosition: PropTypes.string,
-    paddlePrevious: customPropTypes.itemShorthand,
-    thumbnails: PropTypes.bool,
-  };
-
-  static autoControlledProps = ['activeIndex'];
-
-  static defaultProps = {
-    accessibility: carouselBehavior as Accessibility,
-    paddlePrevious: {},
-    paddleNext: {},
-  };
-
-  static Item = CarouselItem;
-  static Navigation = CarouselNavigation;
-  static NavigationItem = CarouselNavigationItem;
-  static Paddle = CarouselPaddle;
-
-  static getAutoControlledStateFromProps(props: CarouselProps, state: CarouselState) {
-    const { items } = props;
-    const { itemIds } = state;
-
-    if (!items) {
-      return null;
-    }
+  });
 
-    return {
-      itemIds: items.map((item, index) => getOrGenerateIdFromShorthand('carousel-item-', item, itemIds[index])),
-    };
-  }
+  const { classes } = useStyles<CarouselStylesProps>(Carousel.displayName, {
+    className: carouselClassName,
+    mapPropsToStyles: () => ({
+      shouldFocusContainer,
+      isFromKeyboard,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  componentWillUnmount() {
-    this.focusItemAtIndex.cancel();
-  }
+  const paddleNextRef = React.createRef<HTMLElement>();
+  const paddlePreviousRef = React.createRef<HTMLElement>();
 
-  actionHandlers = {
-    showNextSlideByKeyboardNavigation: e => {
-      e.preventDefault();
-      this.showNextSlide(e, true);
-    },
-    showPreviousSlideByKeyboardNavigation: e => {
-      e.preventDefault();
-      this.showPreviousSlide(e, true);
-    },
-    showNextSlideByPaddlePress: e => {
-      e.preventDefault();
-      this.showNextSlide(e, false);
-      this.handleNextPaddleFocus();
-    },
-    showPreviousSlideByPaddlePress: e => {
-      e.preventDefault();
-      this.showPreviousSlide(e, false);
-      this.handlePreviousPaddleFocus();
-    },
-  };
+  const focusItemAtIndex = _.debounce((index: number) => {
+    itemRefs[index].current?.focus();
+  }, 400);
 
-  getInitialAutoControlledState(): CarouselState {
-    return {
-      activeIndex: 0,
-      prevActiveIndex: -1,
-      ariaLiveOn: false,
-      itemIds: [] as string[],
-      shouldFocusContainer: false,
-      isFromKeyboard: false,
+  React.useEffect(() => {
```

Comment:
It's not equal to `getAutoControlledStateFromProps` as it's executed before render

Reply:
Will solve in separated PR.